ODT new generation

TeXitoi opened the pull request:
WiP ODT new generation

TeXitoi committed:
ed: read zonal stop points to ed database

TeXitoi committed:
type: create the stop_points_by_area for ODT

antoine-de commented on source/ed/connectors/fusio_parser.cpp:
```diff
@@ -275,6 +287,16 @@ void StopTimeFusioHandler::handle_line(Data& data, const csv_row& row, bool is_f
+        T g;
+        boost::geometry::read_wkt(s, g);
+        return g;
+    } catch (const boost::geometry::read_wkt_exception&) {
```
no log ?

TeXitoi replied:
logged elsewhere

antoine-de commented on source/ed/connectors/fusio_parser.cpp:
```diff
@@ -282,13 +304,13 @@ void GeometriesFusioHandler::init(Data&) {
+            data.shapes[row.at(geometry_id_c)] = {*g};
+        } else if (const auto g = read_wkt<nt::MultiPolygon>(row.at(geometry_wkt_c))) {
+            data.areas[row.at(geometry_id_c)] = *g;
+        }
```
I think you should add the log here

antoine-de commented on source/type/geographical_coord.h:
```diff
@@ -136,3 +139,25 @@ typedef std::vector<LineString> MultiLineString;
+namespace boost { namespace serialization {
+    template<class Archive>
+    void serialize(Archive& ar, navitia::type::Polygon& poly, const unsigned int) {
+        ar & boost::serialization::make_nvp("outer", poly.outer());
```
you don't have to bother with the nvp, it's not handled anywhere :smile:

TeXitoi replied:
That's copy and past from the web ;)

antoine-de commented on source/type/tests/multi_polygon_map_test.cpp:
```diff
@@ -0,0 +1,55 @@
+
+using namespace navitia::type;
+
+BOOST_AUTO_TEST_CASE(inserting_diamon_and_find) {
```
I think you should add a bit more tests (with more than one shape for example)

antoine-de commented on source/type/tests/multi_polygon_map_test.cpp:
```diff
@@ -0,0 +1,55 @@
+    BOOST_REQUIRE(!search.empty());
+    BOOST_CHECK_EQUAL(std::get<2>(*search.begin()), 42);
+
+    for (const auto& e: map) {}
```
debug

TeXitoi committed:
ed: add log when we can't read a geometry

TeXitoi committed:
type: rewrite MultiPolygonMap with RTree

TeXitoi committed:
routing: using is_zonal

TeXitoi committed:
ed: fix reading areas in stop points

TeXitoi committed:
raptor api: use zonal stop points

TeXitoi committed:
jormungandr + routing: handle additional_informations in Journey

antoine-de commented on source/type/multi_polygon_map.h:
```diff
@@ -0,0 +1,93 @@
+            }
+            return true;
+        };
+        rtree.Search(coord, coord, callback, &ctx);
```
ouch :see_no_evil:

TeXitoi replied:
Yeah, this r:deciduous_tree: :no_good:

antoine-de replied:
so now there is the new debian stable you'll change it back to boost::geometry rtree ? :+1:

antoine-de commented on source/type/tests/multi_polygon_map_test.cpp:
```diff
@@ -0,0 +1,54 @@
+    BOOST_CHECK(search.empty());
+    search = map.find(GeographicalCoord(4, 4));
+    BOOST_REQUIRE_EQUAL(search.size(), 1);
+    BOOST_CHECK_EQUAL(search.at(0), 42);
```
can you add test with more elements  ?

TeXitoi replied:
Done with the Beatles

TeXitoi committed:
type: more test of MultiPolygonMap with the Beatles

TeXitoi committed:
routing: filter stop_date_times that are estimated

TeXitoi committed:
routing: update geojson creation for odt

Tests on CO2 are changed because we now use the geojson (and thus the shape
if present) to compute the length of the pt section.

TeXitoi committed:
ed: manage NTFSv0.3

TeXitoi committed:
raptor: filter bad solution before adding them in the pool

TeXitoi committed:
routing: manage stop times to print in the solutions directly in make_path(Journey)

TeXitoi committed:
lots of cleanup and refacto arround VehicleJourneyType removal

TeXitoi committed:
ed: don't segfault on empty vj

TeXitoi committed:
type: rename the fields of OdtLevel_e

TeXitoi committed:
jormungandr: remove the need of VehicleJourneyType

antoine-de commented on source/ed/connectors/fusio_parser.cpp:
```diff
@@ -36,9 +36,15 @@ namespace ed { namespace connectors {
 
 void AgencyFusioHandler::init(Data& data) {
     AgencyGtfsHandler::init(data);
+    if (id_c == -1) { id_c = csv.get_pos_col("network_id"); }
```
i think there is a is_valid(idx_c) helper somewhere

TeXitoi replied:
There isn't.

antoine-de commented on source/ed/connectors/fusio_parser.cpp:
```diff
@@ -1121,9 +1151,14 @@ ed::types::PhysicalMode* GtfsData::get_or_create_default_physical_mode(Data & da
     parse<GeometriesFusioHandler>(data, "geometries.txt");
-    parse<AgencyFusioHandler>(data, "agency.txt", true);
+
+    // TODO: we need at least one of agency.txt or networks.txt
```
I think this TODO is quite useful

antoine-de commented on source/jormungandr/jormungandr/interfaces/common.py:
```diff
@@ -29,11 +29,11 @@
 pb_odt_level = {
-    'none': type_pb2.odt_none,
-    'mixt': type_pb2.mixt,
+    'none': type_pb2.scheduled,
```
remove if useless, else update

antoine-de commented on source/jormungandr/jormungandr/interfaces/v1/fields.py:
```diff
@@ -181,6 +181,15 @@ def output(self, key, obj):
         return str.lower(enum[getattr(obj, key)].name)
 
 
+class PbEnum(fields.Raw):
```
could you add a comment how to use it?

TeXitoi replied:
done

antoine-de commented on source/routing/raptor_api.cpp:
```diff
@@ -71,25 +72,46 @@ static void add_coord(const type::GeographicalCoord& coord, pbnavitia::Section*
-        }
-        return;
-    }
+    if (stop_times.empty()) { return; }
```
lash1 kom!

TeXitoi replied:
done

TeXitoi committed:
jormungandr: fix odt_level

TeXitoi committed:
ed: import only one file for ntfs compatibility

TeXitoi committed:
jormungandr: add comment to PbEnum

TeXitoi committed:
routing: add comments to fill_shape()

TeXitoi committed:
jormungandr: fix new_default
